spec/controllers/categories_controller_spec.rb

```diff
@@ -1,187 +1,204 @@
-        sign_in FactoryGirl.create(:admin)
-      end
-      context 'with valid attributes' do
-        before(:each) do
-          post :create, category: FactoryGirl.attributes_for(:category)
-        end
-        it 'saves a new category to the database' do
-          expect do
-            post :create, category: FactoryGirl.attributes_for(:category)
-          end.to change(Category, :count).by(1)
-        end
-        it { is_expected.to redirect_to(Category.last) }
-        it { is_expected.to set_flash }
-      end
-      context 'with invalid attributes' do
-        before(:each) do
-          post :create,
-               category: FactoryGirl.attributes_for(:category, name: nil)
+    context 'user is admin' do
+      before { mock_user_sign_in(mock_user(:admin)) }
+      context 'successful save' do
+        let!(:cat) { FactoryGirl.build_stubbed(:category) }
+        before do
+          allow(Category).to receive(:new).and_return(cat)
+          allow(cat).to receive(:save).and_return(true)
+          post :create, category: { name: 'Name' }
         end
-        it 'fails to save a new category' do
-          expect do
-            post :create,
-                 category: FactoryGirl.attributes_for(:category, name: nil)
-          end.not_to change(Category, :count)
+        it { is_expected.to set_flash[:notice] }
+        it { is_expected.to redirect_to(cat) }
+      end
+      context 'unsuccessful save' do
+        let!(:cat) { CategoryMock.new }
+        before do
+          allow(Category).to receive(:new).and_return(cat)
+          allow(cat).to receive(:save).and_return(false)
+          post :create, category: { name: 'Name' }
         end
-        it { is_expected.to set_flash }
+        it { is_expected.to set_flash[:error] }
         it { is_expected.to render_template(:new) }
       end
     end
-    context 'not admin' do
-      it 'should redirect to root url' do
-        sign_in FactoryGirl.create(:user)
-        post :create, category: FactoryGirl.attributes_for(:category)
-        expect(response).to redirect_to(root_url)
+    context 'user is not admin' do
+      before do
+        mock_user_sign_in
+        post :create, category: { name: 'Name' }
       end
+      it_behaves_like 'redirected request'
     end
   end
-  describe 'GET edit' do
+
+  describe 'PUT update' do
     context 'is admin' do
-      before(:each) do
-        sign_in FactoryGirl.create(:admin)
-        get :edit, id: @category
-      end
-      it 'should set @category to the selected category' do
-        expect(assigns(:category)).to eq(@category)
+      before { mock_user_sign_in(mock_user(:admin)) }
+      context 'successful update' do
+        let!(:cat) { FactoryGirl.build_stubbed(:category) }
+        before do
+          allow(Category).to receive(:find).with(cat.id.to_s).and_return(cat)
+          allow(cat).to receive(:update_attributes).and_return(true)
+          put :update, id: cat.id, category: { id: 2 }
```

Comment:
The hash is a stand-in for the attributes to be updated--I'll pull it out to a separate variable so it's more clear